test/Microsoft.SqlTools.ServiceLayer.UnitTests/QueryExecution/DataStorage/SaveAsExcelFileStreamWriterHelperTests.cs

```diff
@@ -0,0 +1,220 @@
+            Assert.Equal("Z1", LastWrittenReference);
+            manager.WriteAndIncreaseColumnReference();
+            Assert.Equal("AA1", LastWrittenReference);
+        }
+        [Fact]
+        public void ReferenceZZ1()
+        {
+            var xmlWriter = _xmlWriterMock.Object;
+            var manager = new SaveAsExcelFileStreamWriterHelper.ReferenceManager(xmlWriter);
+            manager.WriteAndIncreaseRowReference();
+
+            for (int i = 0; i < 27 * 26 - 1; i++)
+            {
+                manager.IncreaseColumnReference();
+            }
+            manager.WriteAndIncreaseColumnReference();
+            Assert.Equal("ZZ1", LastWrittenReference);
+            manager.WriteAndIncreaseColumnReference();
+            Assert.Equal("AAA1", LastWrittenReference);
+        }
+        [Fact]
+        public void ReferenceXFD()
+        {
+            var xmlWriter = _xmlWriterMock.Object;
+            var manager = new SaveAsExcelFileStreamWriterHelper.ReferenceManager(xmlWriter);
+            manager.WriteAndIncreaseRowReference();
+
+            for (int i = 0; i < 16384 - 1; i++)
+            {
+                manager.IncreaseColumnReference();
+            }
+            //The 16384 should be the maximal column and not throw
+            manager.AssureColumnReference();
+            manager.WriteAndIncreaseColumnReference();
+            Assert.Equal("XFD1", LastWrittenReference);
+            var ex = Assert.Throws<SaveAsExcelFileStreamWriterHelper.ExporterException>(
+                () => manager.AssureColumnReference());
+            Assert.Contains("max column number is 16384", ex.Message);
+        }
+        [Fact]
+        public void ReferenceRowReset()
+        {
+            var xmlWriter = _xmlWriterMock.Object;
+            var manager = new SaveAsExcelFileStreamWriterHelper.ReferenceManager(xmlWriter);
+            manager.WriteAndIncreaseRowReference();
+            Assert.Equal(1, LastWrittenRow);
+            manager.WriteAndIncreaseColumnReference();
+            Assert.Equal("A1", LastWrittenReference);
+            manager.WriteAndIncreaseColumnReference();
+            Assert.Equal("B1", LastWrittenReference);
+
+            //add row should reset column reference 
+            manager.WriteAndIncreaseRowReference();
+            Assert.Equal(2, LastWrittenRow);
+            manager.WriteAndIncreaseColumnReference();
+            Assert.Equal("A2", LastWrittenReference);
+        }
+
+        [Fact]
+        public void AddRowMustBeCalledBeforeAddCellException()
+        {
+            var xmlWriter = _xmlWriterMock.Object;
+            var manager = new SaveAsExcelFileStreamWriterHelper.ReferenceManager(xmlWriter);
+
+            var ex = Assert.Throws<SaveAsExcelFileStreamWriterHelper.ExporterException>(
+                () => manager.WriteAndIncreaseColumnReference());
+            Assert.Contains("AddRow must be called before AddCell", ex.Message);
+        }
+
+        //This test is too slow
```

Comment:
I would suggest removing the commented out code - it sounds like you don't intend to add back